src/Microsoft.ML.Data/Transforms/FeatureContributionCalculationTransformer.cs

```diff
@@ -266,9 +227,57 @@ private Delegate GetValueGetter<TSrc>(DataViewRow input, int colSrc)
     }
 
     /// <summary>
-    /// Estimator producing a FeatureContributionCalculatingTransformer which scores the model on an input dataset and
-    /// computes model-specific contribution scores for each feature.
+    /// Computes model-specific per-feature contributions to the score of each input vector.
+    /// See the list of currently supported models below.
     /// </summary>
+    /// <remarks>
+    /// <format type="text/markdown"><![CDATA[
+    ///
+    /// ###  Estimator Characteristics
+    /// |  |  |
+    /// | -- | -- |
+    /// | Does this estimator need to look at the data to train its parameters? | No |
+    /// | Input column data type | Vector of floats |
+    /// | Output column data type | Vector of floats |
+    ///
+    /// <para>
+    /// Scoring a dataset with a trained model produces a score, or prediction, for each example. To understand and explain these predictions
+    /// it can be useful to inspect which features influenced them most significantly. This transformer computes a model-specific
+    /// list of per-feature contributions to the score for each example. These contributions can be positive (they make the score higher) or negative
+    /// (they make the score lower).
+    /// </para>
+    /// <para>
+    /// Feature Contribution Calculation is currently supported for the following models:
+    ///     Regression:
+    ///         OrdinaryLeastSquares, StochasticDualCoordinateAscent (SDCA), OnlineGradientDescent, PoissonRegression,
+    ///         GeneralizedAdditiveModels (GAM), LightGbm, FastTree, FastForest, FastTreeTweedie
+    ///     Binary Classification:
+    ///         AveragedPerceptron, LinearSupportVectorMachines, LogisticRegression, StochasticDualCoordinateAscent (SDCA),
+    ///         StochasticGradientDescent (SGD), SymbolicStochasticGradientDescent, GeneralizedAdditiveModels (GAM),
+    ///         FastForest, FastTree, LightGbm
+    ///     Ranking:
+    ///         FastTree, LightGbm
+    /// </para>
+    /// <para>
+    /// For linear models, the contribution of a given feature is equal to the product of feature value times the corresponding weight. Similarly,
+    /// for Generalized Additive Models (GAM), the contribution of a feature is equal to the shape function for the given feature evaluated at
+    /// the feature value.
+    /// </para>
+    /// <para>
+    /// For tree-based models, the calculation of feature contribution essentially consists in determining which splits in the tree have the most impact
```

Comment:
This gives a good description of tree based models. Worth mentioning how it works for the other models?

Reply:
The description for linear models and GAM is above. Do you think it should be more detailed?

---
In reply to: [276423670](https://github.com/dotnet/machinelearning/pull/3384#discussion_r276423670) [](ancestors = 276423670)